src/main/java/igrad/logic/commands/requirement/RequirementAssignCommand.java

```diff
@@ -0,0 +1,90 @@
+ */
+public class RequirementAssignCommand extends RequirementCommand {
+    public static final String COMMAND_WORD = REQUIREMENT_COMMAND_WORD + "assign";
+
+    public static final String MESSAGE_DETAILS = COMMAND_WORD + ": Edits the requirement. "
+        + "Existing requirement will be overwritten by the new name and/or credits.\n";
+
+    public static final String MESSAGE_USAGE = "Parameter: "
+        + "[" + PREFIX_NAME + "NEW_NAME] "
+        + "[" + PREFIX_CREDITS + "NEW_CREDITS]\n"
+        + "Example: " + COMMAND_WORD + " Unrestrained Elves "
+        + PREFIX_NAME + "Unrestricted Electives";
+
+    public static final String MESSAGE_REQUIREMENT_NO_MODULES = "There must be at least one modules assigned.";
+
+    public static final String MESSAGE_MODULES_NON_EXISTENT =
+        "Not all Modules exist in the system. Please try other modules.";
+
+    public static final String MESSAGE_MODULES_ALREADY_EXIST_IN_REQUIREMENT =
+        "Some Modules already exists in this requirement. Please try other modules.";
+    public static final String MESSAGE_REQUIREMENT_ASSIGN_MODULE_SUCCESS = "Modules assigned under Requirement: %1$s";
+
+    private Name requirementName;
+    private List<ModuleCode> moduleCodes;
+
+    public RequirementAssignCommand(Name requirementName, List<ModuleCode> moduleCodes) {
+        requireAllNonNull(requirementName, moduleCodes);
+
+        this.requirementName = requirementName;
+        this.moduleCodes = moduleCodes;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        // Retrieve the requirement in question that we want to assign modules under..
+
+        // First check if the requirement exists in the course book
+        Requirement requirementToAssign = model.getRequirementByName(requirementName)
+            .orElseThrow(() -> new CommandException(MESSAGE_REQUIREMENT_NON_EXISTENT));
+
+        Requirement editedRequirement = new Requirement(requirementToAssign);
+
+        final List<Module> modulesToAssign = model.getModulesByModuleCode(moduleCodes);
+
+        // First check, if all modules (codes) are existent modules in the course book (they should all be)
+        if (modulesToAssign.size() < moduleCodes.size()) {
+            throw new CommandException(MESSAGE_MODULES_NON_EXISTENT);
+        }
+
+        // Now check, if any modules specified are existent in the requirement (they should not)
+        if (editedRequirement.hasModule(modulesToAssign)) {
+            throw new CommandException(MESSAGE_MODULES_ALREADY_EXIST_IN_REQUIREMENT);
+        }
+
+        // Finally if everything alright, we can actually then assign the specified modules under this requirement
+        int creditsAdded = editedRequirement.addModules(modulesToAssign);
+
+        Credits newCreditsFulfilled = editedRequirement.getNewCreditsFulfilled(creditsAdded);
+
+        editedRequirement = new Requirement(editedRequirement.getName(),
```

Comment:
i think this part like a bit weird cos you created a new requirement alr but then over here you recreate the new requirement
i think what you can do is just have your requirement fields out like list modules name and credits, then after everything is modified then you create a new requirement object

Reply:
oh yeah thanks yijie. The most updated code doesnt' look like this, I've changed it back to be consistent with `RequirementEditCommand`.

Reply:
same as below I was under the impression that we need to validate if requirements is full (i.e, adding too many mods under requirement)

Reply:
also I think I got a wrong impression that `creditsFulfilled` (in the `Credits`) class is supposed to be updated each time we assign `Modules` to `Requirement`. but  after you explain to me just now, I was clearer on that, thanks!